pallets/identity/src/lib.rs

```diff
@@ -1159,6 +1162,21 @@ impl<T: Trait> Module<T> {
         None
     }
 
+    pub fn has_valid_kyc(claim_for: IdentityId) -> Option<IdentityId> {
+        let trusted_kyc_providers = T::KycServiceProviders::get_members();
+        for trusted_kyc_provider in trusted_kyc_providers {
```

Comment:
This increases in complexity as the number of trusted KYC providers increase. Since this function will be used often, it might be worth storing this special KYC claim separately as well so that it can always be fetched in O(1).

Reply:
If we want to optimize more then It would be better to add some mapping like 
```
// (Identity of the user, claim key) => trusted SP list who provides the claim value.
map (Identity, claim_key) => Vec<IdnetityId>

```
It will work as a generic solution for every claim. Instead of iterating over all trusted KYC providers. 

AFAIK trusted KYC provider set will be small it never be 1000 SP or more.

Reply:
I don't expect us to have more than a handful of CDD providers. Certainly this will be the case on launch, and will grow very slowly over time I expect. This may be something to optimise down the line rather than now.

Reply:
I've create a new ticket for that: https://polymath.atlassian.net/browse/MESH-736 
We could also evaluate the capability of removing expired CDD claims.